### PR TITLE
DCS-838 extracted select court functionality from request booking file

### PR DESCRIPTION
**Pull request by louisenorris**

### CODE REVIEW

File: @types/whereaboutsApi/index.d.ts [hunk 1]
```diff
@@ -1,6 +1,11 @@
 declare module 'whereaboutsApi' {
   export type CourtLocations = { courtLocations: string[] }
 
+  export type Court = {
+    value: string
+    text: string
+  }
```
Comment (andrewrlee) [inline @types/whereaboutsApi/index.d.ts:7]: these files in `@types` are for apis. Can put this in `./model`

File: backend/routes/requestBooking/selectCourtController.test.ts [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            postAppointmentRequired: 'yes',
+          },
+        ],
+      })
+
+      await controller.view()(req, res, null)
+
+      expect(res.render).toHaveBeenCalledWith(
+        'requestBooking/selectCourt.njk',
+        expect.objectContaining({
```
Comment (andrewrlee) [inline backend/routes/requestBooking/selectCourtController.test.ts:107]: meh.. not the end of the world but would avoid using this `objectContaining` unless you have a specific need to.
This is more for when you have extra fields you don't want to specifically include in your test (generated timestamps that might be difficult to specify in a test etc..). 

It doesn't have a nice comparison view so makes debugging issues more difficult.

File: backend/routes/requestBooking/selectCourtController.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        const preCourtHearingStartTime = moment(startTime, DATE_TIME_FORMAT_SPEC).subtract(20, 'minute')
+        const preCourtHearingEndTime = moment(startTime, DATE_TIME_FORMAT_SPEC)
+        return `${Time(preCourtHearingStartTime)} to ${Time(preCourtHearingEndTime)}`
+      }
+
+      const getPostCourtHearingStartAndEndTime = () => {
+        if (postAppointmentRequired !== 'yes') return 'Not required'
+        const postCourtHearingStartTime = moment(endTime, DATE_TIME_FORMAT_SPEC)
+        const postCourtHearingEndTime = moment(endTime, DATE_TIME_FORMAT_SPEC).add(20, 'minute')
+        return `${Time(postCourtHearingStartTime)} to ${Time(postCourtHearingEndTime)}`
```
Comment (andrewrlee) [inline backend/routes/requestBooking/selectCourtController.ts:45]: We have some helper functions in `bookingTimes` for this. 

could use `getPostAppointmentInterval`, maybe something like: 

```
const [postStart, postEnd] = getPostAppointmentInterval(moment(endTime, DATE_TIME_FORMAT_SPEC))
return `${postStart} to ${postEnd}`
```

there are similar methods for pre which take the `startTime`.

File: @types/whereaboutsApi/index.d.ts (hunk 1 above)
Comment (louisenorris) [reply @types/whereaboutsApi/index.d.ts:7]: Aha, cool, have moved this to` model.ts` as suggested.

File: backend/services/locationService.ts [hunk 4]
```diff
@@ -1,27 +1,41 @@
...
+    return prisons.filter(prison => app.videoLinkEnabledFor.includes(prison.agencyId)).map(this.transformPrison)
+  }
 
-    return prisons
-      .filter(prison => app.videoLinkEnabledFor.includes(prison.agencyId))
-      .map(vlp => ({
-        agencyId: vlp.agencyId,
-        description: vlp.formattedDescription || vlp.description,
-      }))
+  public async getVideoLinkCourtLocations(context: Context): Promise<Court[]> {
```
Comment (andrewrlee) [inline backend/services/locationService.ts:34]: Can we just call this `getVideoLinkEnabledCourts`? 

Locations generally refers to places inside a place, not the place itself.

File: backend/routes/requestBooking/selectCourtController.test.ts (hunk 2 above)
Comment (louisenorris) [reply backend/routes/requestBooking/selectCourtController.test.ts:107]: removed objectContaining as suggested.

File: backend/services/locationService.ts (hunk 4 above)
Comment (louisenorris) [reply backend/services/locationService.ts:34]: Renamed as suggested.

File: backend/routes/requestBooking/selectCourtController.ts (hunk 3 above)
Comment (louisenorris) [reply backend/routes/requestBooking/selectCourtController.ts:45]: Amends made - will push up changes shortly.

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**